src/main/sensors/gyro.c

```diff
@@ -788,6 +788,9 @@ static FAST_CODE void gyroUpdateSensor(gyroSensor_t *gyroSensor, timeUs_t curren
         // still calibrating, so no need to further process gyro data
         return;
     }
+	
+    //update accgyro sensor temperature    
+    gyroReadTemperature();
```

Comment:
This shouldn't be done in the main gyro loop, it can be done in a loop with a much lower frequency.

Reply:
I for sure see your point , in no way do we need temp updates for each gyro loop , Even 5hz sounds like to much?

Personally id maybe only check it every 5-10 seconds into a 5 point hysteresis average. when would a quad go +-20 degrees in 30 seconds?